Fixes two issues with make clean

thedoc31 opened the pull request:
Yesterday, I built haproxy 2.8.5 using `make MAINVERSION=2.8`, which downloaded `haproxy-2.8.5.tar.gz` into `./SOURCES`. Today, 2.8.6 was released so I wanted to build it, but start fresh. I issued `make clean`, but the code tried to remove `haproxy-2.4.25.tar.gz` which was obviously incorrect, *and* the `haproxy-2.8.5.tar.gz` file remained. `make clean MAINVERSION=2.8` did not work either, because it wants to remove `haproxy-2.8.6.tar.gz` (the current upstream version) and nothing else. 

The way `make clean` is configured now, it will only try to remove whatever the current upstream version of `MAINVERSION` is rather than removing any/all versions which might happen to be in `./SOURCES`. This could also get messy for people who might be using the same build box for multiple `MAINVERSION` versions, say `2.7` and `2.8`. 

Therefore, for `make clean` to work as I would expect it to, I propose we remove all `./SOURCES/haproxy-*.tar.gz` archives.

Lastly, the git diff is showing I removed the last line feed from EOF but I didn't - it's still there in code. I am not sure why git is reporting it as a change; I tried to put the line feed back, but just got an empty commit `a364d0f`. 🙄 Apologies for the mess.

thedoc31 committed:
make clean should really clean

thedoc31 committed:
Put back trailing linefeed at EOF

thedoc31 commented:
@DBezemer not sure if you saw this, so ping!

DBezemer commented:
@thedoc31 thanks for the changes, I'm considering that we could empty out the whole SOURCES dir instead as multiple other archives could be placed there as well when building with some options, any comment/concern to that?

thedoc31 commented:
> @thedoc31 thanks for the changes, I'm considering that we could empty out the whole SOURCES dir instead as multiple other archives could be placed there as well when building with some options, any comment/concern to that?

None at all, I was conservative in case there was something else I hadn't thought of. I'll update the PR.

thedoc31 committed:
Remove the entire SOURCES folder

Using DBezemer's suggestion to remove entire SOURCES folder to be extra sure it's starting fresh

thedoc31 commented:
@DBezemer Done! Have a look.

DBezemer commented:
@thedoc31 I just realized clearing out SOURCES will also remove the files provided by the repo, so I think removing all .tar.gz files only is a better idea so it clears up both the LUA archive and the HAP archive. Apologies for the roundabout on this :)

```
rm -f SOURCES/*.tar.gz
```

thedoc31 commented:
> @thedoc31 I just realized clearing out SOURCES will also remove the files provided by the repo, so I think removing all .tar.gz files only is a better idea so it clears up both the LUA archive and the HAP archive. Apologies for the roundabout on this :)
> 
> ```
> rm -f SOURCES/*.tar.gz
> ```

No problem, thanks!
